### PR TITLE
Nayonika/unit patch endpoint

### LINKED ISSUE #80: CRUD 'language/unit'
**Is your feature request related to a problem? Please describe.**
We need endpoints for creating, reading, updating, and deleting units.

**Describe the solution you'd like**
The endpoints can be: 
- POST 'language/unit' for creating a new unit
- GET 'language/unit' for getting unit data and all of the lessons that belong to it.
- PUT 'language/unit' for updating a unit's name, description, and/or order.
- DELETE 'language/unit' for deleting a unit

**Make sure you understand what [unit data we need to store in MongoDB.](https://github.com/hack4impact-uiuc/7000-languages/blob/dev/api/src/models/language.js). And, don't forget to write tests/consider edge cases!**

Take a look at the [issues for `CRUD 'language/courses'`](https://github.com/hack4impact-uiuc/7000-languages/issues/7) for more ideas, hints, and code.

**Resources**
Take a look at the [Mongoose documentation](https://mongoosejs.com/docs/models.html) for more clarification.

### PR DESCRIPTION
**Pull request by nroy20**
## Description

Fixes #80 patch

### CODE REVIEW

File: api/src/api/language/unit.js [hunk 1]
```diff
@@ -41,6 +42,31 @@ router.get(
...
+router.patch(
+  '/:id',
+  requireAuthentication,
+  requireLanguageAuthorization,
+  errorWrap(async (req, res) => {
+    let updates = req.body;
+    if ('course_id' in updates) {
+      updates = _.omit(updates, ['course_id']);
+    }
+    if ('order' in updates) {
```
Comment (ashayp22) [inline api/src/api/language/unit.js:54]: _order instead of order

**Review by ashayp22 — CHANGES_REQUESTED** [review]
Looks good to me, left one inline comment
